package/network/config/netifd/files/lib/netifd/dhcp.script

```diff
@@ -19,12 +19,12 @@ setup_interface () {
 	# TODO: apply $broadcast
 
 	local ip_net
-	eval "$(ipcalc.sh "$ip/$mask")";ip_net="$NETWORK"
+	ip_net="$(ipcalc.sh "$ip/$mask" | sed -nE 's/^NETWORK=//p')"
```

Comment:
The reason I touched the code and introduced the subshell were comments the 'leaky' nature of eval.

```
ipcalc() {
    set -- $(ipcalc.sh "$@")
    [ $? -eq 0 ] && export -- "$@"
}
..
ipcalc "$ip/$mask" && ip_net="$NETWORK"
```
I like the added safety but tend to err more on the 'limit side-effects to subshell'-side of things.
How about `ip_net="$(ipcalc "$ip/$mask" && echo "$NETWORK")"`?

There we'd have a subshell to limit side-effects and no eval either

Reply:
There’s no risk of command injection if you just use the `export`

Reply:
> There we'd have a subshell to limit side-effects and no eval either

You’re misunderstanding how `$(…)` works. For a local shell function, it doesn’t create a subshell. It doesn’t need to.

But subshells have a *lot* of overhead and are to be avoided except when there’s no other way.

Reply:
> There’s no risk of command injection if you just use the `export`

That, I understand. I was talking about not accidentally overwriting variables in that context which `$()` does help with, afaics (`: $(q=1234) ; echo "$q"`).

> For a local shell function, it doesn’t create a subshell.

Fascinating!

Reply:
Right. It doesn’t create a subshell because (1) it’s not necessary, (2) it would be extremely limiting in what you could do, and (3) it would be horribly inefficient.

Your example of an assignment inside of `$(…)` is a bit tortured. Why would anyone do that instead of just assigning without it?

If a function is well behaved and declares all of its variables as `local` then there’s no danger of overwriting variables in parents’ scope. So I’m unclear what the concern or hesitation is.

Reply:
i've closed the 'dittos'.
please resolve this if you're happy with the change :-)